datafusion/functions-aggregate-common/src/aggregate/groups_accumulator.rs

```diff
@@ -163,6 +177,50 @@ impl GroupsAccumulatorAdapter {
 
     /// invokes f(accumulator, values) for each group that has values
     /// in group_indices.
+    fn invoke_per_accumulator<F>(
+        &mut self,
+        values: &[ArrayRef],
+        group_indices: &[usize],
+        opt_filter: Option<&BooleanArray>,
+        total_num_groups: usize,
+        f: F,
+    ) -> Result<()>
+    where
+        F: Fn(&mut dyn Accumulator, &[ArrayRef]) -> Result<()> + Copy,
```

Comment:
We can remove this `Copy` and call with `&f`

Reply:
thanks, done